tests/test_pubkey_cmd.py

```diff
@@ -1,16 +1,55 @@
-def test_get_public_key(cmd):
-    pub_key, chain_code = cmd.get_public_key(
-        bip32_path="m/44'/0'/0'/0/0",
-        display=False
-    )  # type: bytes, bytes
-
-    assert len(pub_key) == 65
-    assert len(chain_code) == 32
-
-    pub_key2, chain_code2 = cmd.get_public_key(
-        bip32_path="m/44'/1'/0'/0/0",
-        display=False
-    )  # type: bytes, bytes
-
-    assert len(pub_key2) == 65
-    assert len(chain_code2) == 32
+from application_client.boilerplate_cmd import BoilerplateCommand, Errors
+from application_client.boilerplate_response_unpacker import unpack_get_public_key_response
+from ragger.bip import calculate_public_key_and_chaincode, CurveChoice
+from ragger.backend import RaisePolicy
+from utils import create_simple_nav_instructions, ROOT_SCREENSHOT_PATH
+
+
+# In this test we check that the GET_PUBLIC_KEY works in non-confirmation mode
+def test_get_public_key_no_confirm(backend, firmware):
+    for path in ["m/44'/0'/0'/0/0", "m/44'/0'/0/0/0", "m/44'/0'/911'/0/0", "m/44'/0'/255/255/255", "m/44'/0'/2147483647/0/0/0/0/0/0/0"]:
+        client = BoilerplateCommand(backend)
+        response = client.get_public_key(path=path).data
+        _, public_key, _, chain_code = unpack_get_public_key_response(response)
+
+        ref_public_key, ref_chain_code = calculate_public_key_and_chaincode(CurveChoice.Secp256k1, path=path)
```

Comment:
I think this should be done with a check on `isinstance(backend, SpeculosBackend):` first. As it will propably not work if tests are run on a real device with a different seed.

Reply:
Nice suggestion on the mnemonic for the conftest, I'll add this in the ragger idea box
Do we agree that the current proposal is sufficient for now?

Reply:
If I'm testing an app with my personal device, I expect to be able to run the tests without having to enter my seed. In such situation this test will fail and I'm not sure it's obvious for the user that it is expected. I would have therefore simply bypassed it when running not on Speculos?
But I agree that the current proposal is definitively mergeable!

Reply:
Mmmh true about not being willing to enter your seed on screen.. But if it's a dev device it's not an issue, we need to make it very clear

Reply:
I agree that for dev device there is probably no issue, but that's not what we should recommend without a big warning.